### DISCUSSION TITLE
Save heart as tag - coming from use-cases (reviving #143)

### DISCUSSION THREAD

**Discussion by totti4ever** [deckerst/aves#694] [Ideas]
Hey, I'd like to give Aves the ultimate challenge and roll it out on my wifes phone to replace the Samsung gallery. She is using the heart feature in the S-Gallery a lot.

Now I understand, that the heart is only an internal information in aves and won't be written into the file. That will most probably lead to a problem as we put our images of shared holidays in one common folder, beeing synced by Syncthing. Plus when arrived on the central server also some renaming action is done to have a common file name syntax across all pictures from all sorts of cameras.

And while I guess that the first part of the use-case would still work with aves (as the movement to the shared folder is done from within aves), I guess the latter part will lose the heart information as the filename is changed from a 3rd party, right?

An alternative woould be to use the rating feature as you said in #143. But I guess, my wife wouldn't like to have more than one click to heart/rate/mark a picture, plus she would have to decide on 1-5 stars instead of just one "like".

@deckerst, do you have any idea how to handle this besides hearting pictures only after they've been moved and renamed? Because that would always carry the danger of losing the heart information once it is moved again (folder renamed on server side and synced back to phone for instance).

cheers

P.S.: Most amazing gallery app I can think of!

**Comment by deckerst** [comment]
Thanks for explaining your use case.

The favourite feature is by design not embedded in the file like tags. The disadvantage is what you're pointing out, but there are some advantages, mainly that it does not need editing support (which is limited to a few formats) and toggling it is instantaneous.

You need persistence for your use case, so tags are the real solution. Your suggestion to save the fav state as tag is a solution worth considering (as an option, the default being the current behaviour), but I'd like to avoid adding options for alternate behaviours, if possible.

Another way is to make tagging as easy as possible. Did you know that you could put the `Edit tags` action as a quick action in the viewer, and if you long press on that quick action you get a pop-up with the most recently used tags? Tagging as favourite becomes a long-press-drag-release gesture. It's not a tap, but it's pretty damned close. Plus, you can still apply another of your common tags with the same gesture. Do you think the long press would be too much of a leap for your spouse? I'm still looking for ways to make tagging easier, by the way, so I'm all ears.
